config/initializers/redis_config.rb

```diff
@@ -1,12 +1,9 @@
-rails_root = ENV['RAILS_ROOT'] || File.dirname(__FILE__) + '/../..'
-rails_env = ENV['RAILS_ENV'] || 'development'
-
-# use config/redis.yml to load settings
-redis_config_path = "#{rails_root}/config/redis.yml"
-redis_config = YAML.load(ERB.new(IO.read(redis_config_path)).result)
+require 'redis'
 
 # initialize redis connection
-Resque.redis = redis_config[rails_env]
-
-# pull out the parsed redis driver from resque
-$redis = Resque.redis.instance_eval{@redis}
+if ENV['REDIS_URL']
+  $redis = Redis.new
+else
+  # use config/secrets.yml to load settings
+  $redis = Redis.new(:url=> Rails.application.secrets.redis_url)
```

Comment:
Would just collapse these two into one like so: 

```ruby
Redis.current =  Redis.new(url: ENV['REDIS_URL'] || Rails.application.secrets.redis_url)
```

Would use `Redis.current` over a global ruby variable. Also use ruby 1.9 hash syntax instead of hash rockets ( `=>`)